### DISCUSSION TITLE
Selected (Active) Tab color

### DISCUSSION THREAD

**Discussion by SNike-1** [sumatrapdfreader/sumatrapdf#3764] [Ideas]
Many files are open. Their names are similar, and they don't fit in the tabs. It's hard to figure out which tab is open. I suggest to somehow highlight the active tab. That would be very good. Thanks!

**Comment by kjk** [comment]
That should already be the case in latest pre-release https://www.sumatrapdfreader.org/prerelease

**Comment by SNike-1** [comment]
> That should already be the case in latest pre-release
I didn't notice. 
![изображение](https://github.com/sumatrapdfreader/sumatrapdf/assets/147944305/c71eca22-30ed-4d5f-b28b-f7358f6c72f1)

How to turn active tab highlite on?

**Comment by kjk** [comment]
I guess I have to make the contrast bigger. The color for active tab is different but it looks it should be more different.

**Comment by SNike-1** [comment]
if the active tab is clearly distinguishable among others, it will be very convenient!

**Comment by kjk** [comment]
I've increased the contrast to this:

![SumatraPDF_34xj6alrTF](https://github.com/sumatrapdfreader/sumatrapdf/assets/7935/005028d6-6546-428c-aa2d-495fd1864067)

**Reply by SNike-1** [reply]
Yes, so good. Thanks!

**Reply by hradilik** [reply]
I apologize, but contrast of activet tab is still too low for me (in 3.6.17052 pre-release). Would it be possible to add the ActiveTabBackgroundColor and ActiveTabForegroundColor parameters to the advanced settings? Or just the InvertActiveTabColor parameter? Thank You.

**Comment by username116** [comment]
A suggestion on colors in dark mode: (sorry I'm intruding on a discussion when this isn't exactly the topic)

Make a slight difference between the background colors of the window and the document. This would allow the edges of the document to be seen.
Currently, when document colors are also inverted, these colors (and that of the toolbar) are the same.
Or make these colors customizable in the advanced options, like the light theme? (but it must be some work, it's easy to have ideas).

Screenshots with the current colors, and a screenshot of another program, LibreOffice, as an example:
<details>
<summary>Click to unfold</summary>

![Screenshot dark](https://github.com/sumatrapdfreader/sumatrapdf/assets/108369600/0fbd0da2-f8bb-4fa7-b43a-345ee19d70cc)

![Screenshot light](https://github.com/sumatrapdfreader/sumatrapdf/assets/108369600/94bd8db0-a793-4068-ba0f-789a5c6ec1f0)

![Screenshot LibreOffice](https://github.com/sumatrapdfreader/sumatrapdf/assets/108369600/98a8781a-009b-49f4-b134-524b105cb4d3)

</details>

**Reply by GitHubRulesOK** [reply]
@username116 
SumatraPDF has had an "experimental"  background feature as long as I can remember so you chose what you want within reason
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/ae080e42-e3c9-4110-aeb3-a8c02020c75e)

one to three colours based on `GradientColors = #ff0000 #0000FF` after FixedPageUI [

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/f7036d12-48ff-4d90-aabb-427f37ed79d7)

**Reply by kjk** [reply]
@username116 done in https://github.com/sumatrapdfreader/sumatrapdf/commits/master, will be in next pre-release build

**Reply by username116** [reply]
Thank you very much for your answers.

@GitHubRulesOK I hadn't tried this setting. I tried it, and indeed I can set the background color I want, even in dark mode. Thanks for the tip.

@kjk thanks, I look forward to the next pre-release.

Idea: indicate in the [advanced settings help](https://www.sumatrapdfreader.org/settings/settings) of the next version, for `TextColor`, `BackgroundColor` and `MainWindowBackground` settings, that these settings don't work with dark themes. And for `MainWindowBackground`, that `GradientColors` can be used instead for a setting with dark theme.
This could save time for people like me trying to adjust the dark theme colors.

**Reply by username116** [reply]
Here's the result and how it compares with the previous version.

I don't see any difference with the light theme on my screenshots (although the code 98b86d67c055ffa5847ba63fbe455ca71ba6f145 also changes the lightness with the light theme?).
I would have set -8 instead of +8 with the dark theme, to darken the background of the document a little, instead of lightening it. With the text which is already a little greyed out, it lacks contrast in my eyes. But I got what I wanted, the edges of the sheet are visible. Thanks for the change.

<details>
<summary>Click to unfold</summary>

Before, with the light theme: 

![15722 light](https://github.com/sumatrapdfreader/sumatrapdf/assets/108369600/2d2bf49c-fbfa-4ff2-99b0-2e40cb2e6493)

After : 

![15732 light](https://github.com/sumatrapdfreader/sumatrapdf/assets/108369600/8fca6d92-a89c-4703-8738-8cbb84dde09f)

Before, with the dark theme: 

![15722 dark](https://github.com/sumatrapdfreader/sumatrapdf/assets/108369600/68f6eeb2-1f7e-4aa4-bbfd-636d97f015bb)

After : 

![15732 dark](https://github.com/sumatrapdfreader/sumatrapdf/assets/108369600/a8978a7e-7aed-494f-94a7-d4b4f5fdd140)

</details>

**Reply by kjk** [reply]
If you look at at GetDocumentColors() it treats the light theme in a special way so this code won't execute which is why the light theme didn't change.
